flake.nix

```diff
@@ -1,49 +1,90 @@
-
-  inputs = {
-    flake-compat = {
-      url = "github:edolstra/flake-compat";
-      flake = false;
-    };
-
-    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
-  };
-
-  outputs = { self, nixpkgs, flake-compat, ... }:
+  description = "Scenefx development environment";
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+  outputs =
+    { self, nixpkgs, ... }:
     let
-      pkgsFor = system:
-        import nixpkgs {
-          inherit system;
-          overlays = [ ];
-        };
-
-      targetSystems = [ "aarch64-linux" "x86_64-linux" ];
-    in {
-      devShells = nixpkgs.lib.genAttrs targetSystems (system:
-        let pkgs = pkgsFor system;
-        in {
-          default = pkgs.mkShell {
-            name = "scenefx-shell";
-
-            inputsFrom = [ pkgs.wlroots_0_17 ];
+      mkPackages = pkgs: {
+        scenefx = pkgs.callPackage (
+          { wlroots_0_17, ... }:
+          pkgs.stdenv.mkDerivation {
+            pname = "scenefx";
+            version = "0.1.0-git";
+            src = ./.;
+            outputs = [
+              "out"
+              "lib"
+            ];
 
             nativeBuildInputs = with pkgs; [
-              cmake
+              pkg-config
               meson
+              cmake
               ninja
               scdoc
-              pkg-config
-              hwdata
             ];
 
-            shellHook = with pkgs; ''(
+            buildInputs = with pkgs; [
+              libdrm
+              libxkbcommon
+              pixman
+              libGL # egl
+              mesa # gbm
+              wayland # wayland-server
+              wayland-protocols
+              wlroots_0_17
+            ];
+
+            meta = with pkgs.lib; {
+              description = "A drop-in replacement for the wlroots scene API that allows wayland compositors to render surfaces with eye-candy effects";
+              homepage = "https://github.com/wlrfx/scenefx";
+              license = licenses.mit;
+              platforms = platforms.linux;
+            };
+          }
+        ) { };
+      };
+
+      targetSystems = [
+        "aarch64-linux"
+        "x86_64-linux"
+      ];
+      pkgsFor = system: import nixpkgs { inherit system; };
+      forEachSystem = f: nixpkgs.lib.genAttrs targetSystems (system: f (pkgsFor system));
+    in
+    {
+      overlays = rec {
+        default = insert;
+        override = _: prev: mkPackages prev;
+        insert = _: prev: mkPackages (pkgsFor prev.system);
+      };
+
+      packages = forEachSystem (
+        pkgs: (mkPackages pkgs) // { default = self.packages.${pkgs.system}.scenefx; }
+      );
+
+      devShells = forEachSystem (pkgs: {
+        default = pkgs.mkShell {
+          name = "scenefx-shell";
+
+          inputsFrom = [
+            self.packages.${pkgs.system}.scenefx
+            pkgs.wlroots_0_17
+          ];
+
+          nativeBuildInputs = [
+            pkgs.hwdata # for wlroots
+          ];
```

Comment:
It looks like wlroots 0.17 in nixpkgs already has that input, do we still need it here, if we already have `wlroots_0_17` in `inputsFrom`?

https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/libraries/wlroots/default.nix#L132

Reply:
Maybe not, i carried it over from the legacy code since it had the explicit comment. If it works without I'm happy to make the change!

Reply:
Ah ye, i see Will added that in the SwayFX flake including the comment, so @WillPower3309 what do you think?

Reply:
Yeah for whatever reason I need to explicitly add it or swayfx's wlroots subproject won't compile. I'll try it with and without on this branch and see if it's required here too

Reply:
can confirm this block isn't needed. Weird!

Reply:
otherwise LGTM

Reply:
Cleaned up the devshell!